File: cpp/src/Motion/ServoVelocity.cpp
//#include "ctre/phoenix/Motion/ServoVelocity.h"
//
//namespace ctre { namespace motion {
//
//ServoVelocity::ServoVelocity(ctre::motorcontrol::SmartMotorController *motor,
//		ctre::motorcontrol::SmartMotorController::FeedbackDevice feedbackDevice, float velocity)
//{
//    ctre::Mechanical::SensoredGearbox *temp1 = new ctre::Mechanical::SensoredGearbox(1, motor, feedbackDevice);
//    _gearbox = temp1;
//    _targetVelocity = velocity;
//}
//ServoVelocity::ServoVelocity(ctre::Mechanical::SensoredGearbox *gearbox, float velocity){
//	_gearbox = gearbox;
//    _targetVelocity = velocity;
//}
//void ServoVelocity::Set(float targetSpeed, float maxOutput){
//	_gearbox->ConfigPeakOutputVoltage(maxOutput, -maxOutput);
//	_gearbox->SetControlMode(ctre::motorcontrol::ControlMode::SmartControlMode::kSpeed);
//	_gearbox->Set(targetSpeed);
//}
//void ServoVelocity::OnStart(){
//
//}
//void ServoVelocity::OnStop(){
//    /* stop the talon */
//	_gearbox->SetControlMode(ctre::motorcontrol::ControlMode::kPercentVbus);
//	_gearbox->Set(0);
//}
//bool ServoVelocity::IsDone(){
//	return false;
//}
//void ServoVelocity::OnLoop(){
//	_gearbox->SetControlMode(ctre::motorcontrol::ControlMode::kSpeed);
//	_gearbox->Set(_targetVelocity);
//}
//
//
//}}

File: java/src/com/ctre/phoenix/Motion/ServoVelocity.java
//package com.ctre.phoenix.Motion;
//
//import com.ctre.phoenix.Drive.*;
//import com.ctre.phoenix.MotorControl.*;
//import com.ctre.phoenix.MotorControl.ControlMode.SmartControlMode;
//import com.ctre.phoenix.Mechanical.*;
//
//public class ServoVelocity implements com.ctre.phoenix.ILoopable
//{
//
//    SensoredGearbox Gearbox;
//    double _targetVelocity = 0;
//    
//    /* SmartMotorController Constructor */
//    public ServoVelocity(SmartMotorController motor, SmartMotorController.FeedbackDevice feedbackDevice, double velocity)
//    {
//        SensoredGearbox temp1 = new SensoredGearbox(1, motor, feedbackDevice);
//        Gearbox = temp1;
//        _targetVelocity = velocity;
//    }
//    /* SensoredGearbox Constructor */
//    public ServoVelocity(SensoredGearbox gearbox, double velocity)
//    {
//        Gearbox = gearbox;
//        _targetVelocity = velocity;
//    }
//
//    public void onStart()
//    {
//    }
//
//    public void onStop()
//    {
//        /* stop the talon */
//        Gearbox.SetControlMode(ControlMode.SmartControlMode.kPercentVbus);
//        Gearbox.Set(0);
//    }
//    
//    public void set(double targetSpeed, double maxOut)
//    {
//    	Gearbox.ConfigPeakOutputVoltage(maxOut, -maxOut);
//    	Gearbox.SetControlMode(SmartControlMode.kSpeed);
//    	Gearbox.Set(targetSpeed);
//    }
//
//    public boolean isDone()
//    {
//        return false;
//    }
//
//    public void onLoop()
//    {
//        Gearbox.SetControlMode(ControlMode.SmartControlMode	.kSpeed);
//        Gearbox.Set(_targetVelocity);
//    }
//}